src/libs/actions/Session/index.js

```diff
@@ -452,39 +453,39 @@ const reauthenticatePusher = _.throttle(() => {
 function authenticatePusher(socketID, channelName, callback) {
     Log.info('[PusherAuthorizer] Attempting to authorize Pusher', false, {channelName});
 
-    DeprecatedAPI.Push_Authenticate({
+    // We use makeRequestWithSideEffects here because we need to authorize to Pusher (an external service) each time a user connects to any channel.
+    // eslint-disable-next-line rulesdir/no-api-side-effects-method
+    API.makeRequestWithSideEffects('AuthenticatePusher', {
         socket_id: socketID,
         channel_name: channelName,
         shouldRetry: false,
         forceNetworkRequest: true,
-    })
-        .then((response) => {
-            if (response.jsonCode === CONST.JSON_CODE.NOT_AUTHENTICATED) {
-                Log.hmmm('[PusherAuthorizer] Unable to authenticate Pusher because authToken is expired');
-                callback(new Error('Pusher failed to authenticate because authToken is expired'), {auth: ''});
+    }).then((response) => {
```

Comment:
**nonblocking fussy observation:** if we just leave this indent alone the diff will leave less of an impact

Reply:
I think the new way is better for style but yeah

Reply:
I prefer `.then()` on a new line, but it is a personal preference and not something we have in the style guide.